openpype/hosts/maya/api/lib.py

```diff
@@ -3053,31 +3061,52 @@ def remove_render_layer_observer():
         pass
 
 
-def update_content_on_context_change():
-    """
-    This will update scene content to match new asset on context change
+def update_instances_frame_range():
+    """Update 'frameStart', 'frameEnd', 'handleStart', 'handleEnd' and 'asset'
+    attributes of sets that got one, execpt if instance family is 'render'
     """
-    scene_sets = cmds.listSets(allSets=True)
+    collected_instances = cmds.ls(
+        "*.id",
+        long=True,
+        type="objectSet",
+        recursive=True,
+        objectsOnly=True
+    )
     asset_doc = get_current_project_asset()
     new_asset = asset_doc["name"]
     new_data = asset_doc["data"]
```

Comment:
With the function name refactored to `update_instances_frame_range` the `new_` suffix to this variable seems a bit odd. I feel like we could just refactor these variables to: `asset_name` and `asset_data`.